android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Interactor.kt

```
@@ -42,6 +43,8 @@ abstract class Interactor<C : Parcelable, V : RibView>(
     private val disposables: Disposable?
 ) : LifecycleScopeProvider<InteractorEvent>, LifecycleOwner {
 
+    lateinit var node: Node<V>
```

Do we want to have lateinit node here and crash if it will not be set?

---

Unfortunately it needs to be lateinit, as Node also needs Interactor. It sets the reference in its init block, just the same as with Router. I don't see any other way, please update if you do.

---

Yes, I thought maybe it is better to use Nullable field and later:
`node?.startActivityForResult`?

And here we have circular dependency - Interactor knows about Node and Node knows about Interactor, usually I am resolving it introducing interface to decouple class, but I am not sure that it makes sense to do here.

---

`node?.startActivityForResult` would fail silently. If the reference was not set, it is the fault of the framework, not client code, and as such, it should crash and fail the CI build.

---

yes, make sense. I agree.